### PR TITLE
feat(raw_vehicle_cmd_converter): add steer command conversion with VGR

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

Added conversion from steer angle to handle angle by variable gear ratio.
Both cmd and status are supported. (However, status conversion is not supported for steer_map.)
This is to unify the conversion of control commands, which has been done in vehicle_interface.
Also, status correction will be implemented in the future.

image

## Related links

**Parent Issue:**

- Link

## How was this PR tested?

- psim wtih https://github.com/autowarefoundation/autoware.universe/pull/8415
- unit test
- 
## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: vehicle/autoware_raw_vehicle_cmd_converter/include/autoware_raw_vehicle_cmd_converter/node.hpp [hunk 1]
```diff
@@ -102,16 +108,26 @@ class RawVehicleCommandConverterNode : public rclcpp::Node
...
   bool is_debugging_;
-  bool convert_accel_cmd_;  //!< @brief use accel or not
-  bool convert_brake_cmd_;  //!< @brief use brake or not
-  bool convert_steer_cmd_;  //!< @brief use steer or not
+  bool convert_accel_cmd_;                                             //!< @brief use accel or not
+  bool convert_brake_cmd_;                                             //!< @brief use brake or not
+  std::optional<std::string> convert_steer_cmd_method_{std::nullopt};  //!< @brief method to convert
+  double vgr_coef_a_{0.0};
+  double vgr_coef_b_{0.0};
+  double vgr_coef_c_{0.0};
```
Comment (TakaHoribe) [inline vehicle/autoware_raw_vehicle_cmd_converter/include/autoware_raw_vehicle_cmd_converter/node.hpp:116]: Why are these members necessary?

**Review by TakaHoribe — COMMENTED** [review]
Could you update the README, particularly by adding an image of the steering conversion scheme and clarifying/plotting what kind of value the VGR would be?

File: vehicle/autoware_raw_vehicle_cmd_converter/include/autoware_raw_vehicle_cmd_converter/node.hpp (hunk 1 above)
Comment (kosuke55) [reply vehicle/autoware_raw_vehicle_cmd_converter/include/autoware_raw_vehicle_cmd_converter/node.hpp]: not necessary. it was for before refactoring, so removed them.

**Comment by kosuke55** [comment]
@TakaHoribe 

> Could you update the README, particularly by adding an image of the steering conversion scheme and clarifying/plotting what kind of value the VGR would be?

thanks! I updated readme and scripts for plot.

File: vehicle/autoware_raw_vehicle_cmd_converter/README.md [hunk 2]
```diff
@@ -45,3 +68,4 @@ For ease of calibration and adjustments to the lookup table, an auto-calibration
 ## Limitation
 
 The current feed back implementation is only applied to steering control.
+`
```
Comment (TakaHoribe) [inline vehicle/autoware_raw_vehicle_cmd_converter/README.md:71]:
```suggestion
```

File: vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp [hunk 3]
```diff
@@ -86,11 +109,11 @@ RawVehicleCommandConverterNode::RawVehicleCommandConverterNode(
 
 void RawVehicleCommandConverterNode::publishActuationCmd()
 {
-  if (!current_twist_ptr_ || !control_cmd_ptr_ || !current_steer_ptr_) {
+  if (!current_twist_ptr_ || !control_cmd_ptr_ || !current_steer_ptr_ || !actuation_status_ptr_) {
```
Comment (TakaHoribe) [inline vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp:112]: Some vehicles do not publish actiation_status. This is not necessary for vehicles that do not require the steering conversion. Please make a specific condition for that.

File: vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp [hunk 4]
```diff
@@ -215,6 +241,41 @@ void RawVehicleCommandConverterNode::onControlCmd(const Control::ConstSharedPtr
   control_cmd_ptr_ = msg;
   publishActuationCmd();
 }
+
+void RawVehicleCommandConverterNode::onSteering(const Steering::ConstSharedPtr msg)
```
Comment (TakaHoribe) [inline vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp:245]: Please mention that this subscriber can not be replaced by `take()` method in a comment.

Comment (kosuke55) [reply vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp:245]: Originally it was take(), in which case the topic name must be stated in the declaration. In that case, there would be two identical topics. Therefore, subscriber, which does not require a topic name to be specified at the time of declaration, is used.
In any case, add a comment to the code.

image

Comment (kosuke55) [reply vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp:245]: > In that case, there would be two identical topics.

I mean, this node also need to publish steering topic if converting option is true.

image

File: vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp (hunk 3 above)
Comment (kosuke55) [reply vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp:122]: thanks! fixed in https://github.com/autowarefoundation/autoware.universe/pull/8504/commits/0a660ef011c913acebdc84291856bdea4e366410

File: vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp (hunk 4 above)
Comment (kosuke55) [reply vehicle/autoware_raw_vehicle_cmd_converter/src/node.cpp:263]: added comment why not use polling sub

https://github.com/autowarefoundation/autoware.universe/pull/8504/commits/5b667ccf9f6052a37c888062a623273f2eaf4254

**Review by TakaHoribe — APPROVED** [review]

**Pull request merged**